cluster-api/cmd/controller-manager/Makefile

```diff
@@ -0,0 +1,32 @@
+# Copyright 2018 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+.PHONY: image push dev_image dev_push
+
+PREFIX = gcr.io/k8s-cluster-api
+DEV_PREFIX ?= gcr.io/$(shell gcloud config get-value project)
+NAME = controller-manager
+TAG = 0.0.1
+
+image:
+	docker build -t "$(PREFIX)/$(NAME):$(TAG)" -f ./Dockerfile ../..
+
+push: image
+	docker push "$(PREFIX)/$(NAME):$(TAG)"
```

Comment:
maybe it's just me but i can only seem to push to gcr using gcloud docker rather than just docker.

Reply:
@roberthbailey what is the issue you run into? Perhaps some auth issue?

It seems we should not require gcloud if this is not a GCP specific code.

Reply:
This maybe got lost in my email reply, but `gcloud beta auth configure-docker` will install gcloud as a credential store for docker. This allows `docker push` to work with gcr.io.